packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp

```diff
@@ -520,21 +631,44 @@ YogaLayoutableShadowNode& YogaLayoutableShadowNode::cloneChildInPlace(
     size_t layoutableChildIndex) {
   ensureUnsealed();
 
-  const auto& childNode = *yogaLayoutableChildren_[layoutableChildIndex];
+  const auto& child = yogaLayoutableChildren_.at(layoutableChildIndex);
+  const auto& childNode = *child.directChild;
 
-  // TODO: Why does this not use `ShadowNodeFragment::statePlaceholder()` like
-  // `adoptYogaChild()`?
-  auto clonedChildNode = childNode.clone(
-      {ShadowNodeFragment::propsPlaceholder(),
-       ShadowNodeFragment::childrenPlaceholder(),
-       childNode.getState()});
+  ShadowNode::Unshared clonedChildNode = nullptr;
+  
+  size_t index = 0;
+  for (const auto& node : yogaLayoutableChildren_) {
+    if (node.yogaChild.get() == &childNode) {
+      break;
+    }
+    index++;
+  }
 
+  ShadowNode* result = &*clonedChildNode;
+  
+  // entire path will be cloned for each child
+  if (&*child.directChild != &*child.yogaChild) {
+    clonedChildNode = childNode.cloneTree(child.yogaChild->getFamily(), [&](const ShadowNode& oldNode) {
+      const auto clone = oldNode.clone({.state = oldNode.getState()});
+      result = &*clone;
+      return clone;
+    });
+  } else {
+    // TODO: Why does this not use `ShadowNodeFragment::statePlaceholder()` like
+    // `adoptYogaChild()`?
+    clonedChildNode = childNode.clone(
+        {ShadowNodeFragment::propsPlaceholder(),
+          ShadowNodeFragment::childrenPlaceholder(),
+          childNode.getState()});
```

Comment:
Not really related, but I think the answer is related to https://github.com/facebook/react-native/pull/44796. TL;DR cloning a node with a state placeholder will replace the state in the node with the most recently committed one.